sky/backends/backend_utils.py

```diff
@@ -1968,7 +1972,7 @@ def _update_cluster_status_no_lock(
         ssh_credentials = ssh_credential_from_yaml(handle.cluster_yaml)
         runner = command_runner.SSHCommandRunner(external_ips[0],
                                                  **ssh_credentials)
-        rc, output, _ = runner.run('ray status',
+        rc, output, _ = runner.run(f'{_SET_RAY_ADDRESS}ray status',
```

Comment:
Curious: Why was there an INIT transition? Is it because originally here we queried the user-level ray cluster, and the _count_healthy_nodes_from_ray() below somehow didn't show 1 node being healthy? What's the buggy output from _count_healthy_nodes_from_ray()?

Reply:
The problem is caused by the `_count_healthy_nodes_from_ray` as the user's `ray status` will be as following, which does not contain the node name, i.e. `ray_head_default`/`ray_worker_default`.
```
2023-06-04 05:48:34,791 WARNING services.py:405 -- Found multiple active Ray instances: {'10.128.0.47:6380', '10.128.0.47:7379'}. Connecting to latest cluster at 10.128.0.47:7379. You can override this by setting the `--address` flag or `RAY_ADDRESS` environment variable.
======== Autoscaler status: 2023-06-04 05:48:32.064004 ========
Node status
---------------------------------------------------------------
Healthy:
 1 node_b9d99c9b660a0266a95b423ab4647c52eb3d888299de4f315d4f4375
Pending:
 (no pending nodes)
Recent failures:
 (no failures)

Resources
---------------------------------------------------------------
Usage:
 0.0/4.0 CPU
 0B/8.33GiB memory
 0B/4.17GiB object_store_memory

Demands:
 (no resource demands)
```